### PR TITLE
Implementation of the Badge component

### PR DESCRIPTION
**Pull request by hcafaq**
### Description

Adds Badge component
[Issue Ref](https://heycar.atlassian.net/jira/software/projects/HEYUI/boards/312?selectedIssue=HEYUI-114)
[Design](https://www.figma.com/file/rrvkvQEoVTOHa7MzyKSaoz/OTP-Global-Design-System?node-id=4031%3A102387)

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation change

### How do I test this

- import Badge from '@heycar-uikit/badge';
- <Badge
            symbol="$"
            count="250"
            text="off"
            background={variant}
            showIcon={true}
          />

### Did you remember to take care of the following?

- [x] `npm i` – for new NPM dependencies.
- [x] `npm run lint` - to check for linting issues
- [x] `npm run test` - to run unit tests
- [ ] `npm run test:screenshots` - to run snapshot tests

### New Feature / Bug Fix

- [x] Run unit tests to ensure all existing tests are still passing.
- [x] Add new passing unit tests to cover the code introduced by your pr.

Thanks for contributing!

### CODE REVIEW

File: packages/badge/src/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import cn from 'classnames';
+
+import { BadgeProps } from './Badge.types';
+
+import styles from './styles/default.module.css';
+import styleVariants from './styles/variant.module.css';
+
+function Badge({ color = 'primary', dataTestId, children, text }: BadgeProps) {
```
Comment (derek-heycar) [inline packages/badge/src/Badge.tsx:9]: `forwardRef`
`Component  = 'div'`

Comment (hcafaq) [reply packages/badge/src/Badge.tsx]: done

File: packages/badge/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+## Usage
+
+```tsx
+import badge from '@heycar-uikit/badge';
+
+function App() {
+  return (
+    <badge size="large" color="primary" >
+    </badge>
```
Comment (derek-heycar) [inline packages/badge/README.md:21]: add children

File: packages/badge/src/docs/Component.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+<Meta title="Components/Badge" component={Badge} />
+
+export const BadgeTemplate = args => <Badge {...args} />;
+
+<!-- Canvas -->
+
+<Story
+  name="Badge"
+  args={{
+    leftIcon: <Lorry fontSize="20" />,
```
Comment (derek-heycar) [inline packages/badge/src/docs/Component.stories.mdx:23]: remove leftIcon prop from control board

File: packages/badge/src/styles/variant.module.css [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  --badge-tertiary-background: var(--color-tertiary-500);
+
+  /* warning */
+  --badge-warning-background: var(--color-warning-400);
+
+  /* highlight */
+  --badge-highlight-background: var(--color-highlight-500);
+}
+
+.badge-primary {
```
[the remark this answers is not in the document]
Comment (hcafaq) [reply packages/badge/src/styles/variant.module.css]: done

File: packages/badge/README.md (hunk 2 above)
Comment (hcafaq) [reply packages/badge/README.md:22]: done

File: packages/badge/src/docs/Component.stories.mdx (hunk 3 above)
Comment (hcafaq) [reply packages/badge/src/docs/Component.stories.mdx]: done

File: packages/badge/README.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+To install and save in your package.json dependencies, run the command below using npm:
+
+```bash
+npm install @heycar-uikit/badge
+```
+
+## Usage
+
+```tsx
+import badge from '@heycar-uikit/badge';
```
Comment (derek-heycar) [inline packages/badge/README.md:16]: `import badge from '@heycar-uikit/badge'` => `import Badge from '@heycar-uikit/badge'`

File: packages/badge/README.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+function App() {
+  return (
+    <badge size="large" color="primary">
```
Comment (derek-heycar) [inline packages/badge/README.md:20]: `<Badge...`

File: packages/badge/src/__tests__/Badge.snapshot.test.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import Badge from '../Badge';
+
```
Comment (derek-heycar) [inline packages/badge/src/__tests__/Badge.snapshot.test.tsx:5]: `const defaultChild = '$100 off'`

File: packages/badge/src/__tests__/Badge.test.tsx [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+const dataTestId = 'test-id';
+
+describe('Badge', () => {
+  /**
+   * Attributes tests
+   */
+  describe('Attributes tests', () => {
+    it('should set `data-test-id` attribute', () => {
+      const { getByTestId } = render(
+        <Badge dataTestId={dataTestId}> $250 off </Badge>,
```
Comment (derek-heycar) [inline packages/badge/src/__tests__/Badge.test.tsx:15]: ` $250 off `  to variable

Comment (hcafaq) [reply packages/badge/src/__tests__/Badge.test.tsx]: resolved

File: packages/badge/src/__tests__/Badge.snapshot.test.tsx (hunk 7 above)
Comment (hcafaq) [reply packages/badge/src/__tests__/Badge.snapshot.test.tsx:5]: resolved

File: packages/badge/README.md (hunk 6 above)
Comment (hcafaq) [reply packages/badge/README.md]: resolved

File: packages/badge/README.md (hunk 5 above)
Comment (hcafaq) [reply packages/badge/README.md]: resolved

**Pull request merged**